libcloud/loadbalancer/drivers/gce.py

```diff
@@ -338,9 +338,12 @@ def _node_to_member(self, node, balancer):
         # would be found if it was there.
         if hasattr(node, 'name'):
             member_id = node.name
-            member_ip = node.public_ips[0]
         else:
             member_id = node
+
```

Comment:
Although a correct solution, I think we should solve this by fixing the `Node` object, which should always have a `public_ips` attribute that is empty if there are no public IPs. This has the benefit of us needing to code that only once instead of all over the place.

Also, I would like to see a testcase, so this functionality never gets accidentally removed.